### PR TITLE
Chevron: filter with '*' in front

### PR DESCRIPTION
**Pull request by jukzi**
image

### CODE REVIEW

**Comment by jukzi** [comment]
A second opinion about this change would be appreciated.

**Comment by BeckerWdf** [comment]
> A second opinion about this change would be appreciated.

Sure. Can you please describe the problem you want to fix? "Chevron: filter with '*' in front" is not very speaking..

**Comment by jukzi** [comment]
until now the filter finds only files starting with the String:
image
and would not find others:
image
That behaviour may be OK for Open-Type Dialog which filters from thousands of classes, but there are at max 99 items to filter in the chevron and i often find that i would have liked it to find the `*Test` too

File: bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/internal/workbench/renderers/swt/AbstractTableInformationControl.java [hunk 1]
```diff
@@ -403,6 +403,9 @@ private void setMatcherString(String pattern) {
 			fSearchPattern = null;
 		} else {
 			SearchPattern patternMatcher = new SearchPattern();
+			if (!pattern.startsWith("*")) { //$NON-NLS-1$
+				pattern = "*" + pattern; //$NON-NLS-1$
```
Comment (BeckerWdf) [inline bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/internal/workbench/renderers/swt/AbstractTableInformationControl.java:407]: Change looks good to me.
But one question: Is this class only used for the Cevron-List? If not we may create unwanted side effects..

Comment (jukzi) [reply bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/internal/workbench/renderers/swt/AbstractTableInformationControl.java:407]: As far as i have searched yes. But i wonder how many options those classes have even though they are internal and only used once:
image

**Review by BeckerWdf — APPROVED** [review]

**Comment by jukzi** [comment]
ignoring random failing PartRenderingEngineTests https://github.com/eclipse-platform/eclipse.platform.ui/issues/751

**Pull request merged**